### DISCUSSION TITLE
[RFC] Sync Metaplex custom SPL token auctions with Holaplex/metaplex

### DISCUSSION THREAD

**Discussion by kevinrodriguez-io** [holaplex/metaplex#330] [Ideas]
### Preamble

As part of the extension and mass adoption of crypto currency, including payment platforms that mainly use tokens like USDC, it would be good to allow auction creators to select their own SPL tokens to accept and a given amount of that token, instead of the current supported token which is only SOL. Allowing people to instantly sell NFTs in tokens like USDC, or allowing DAOS to sell NFT merch using their own DAO Tokens will be a nice feature to have for many users.

Initially, holaplex/metaplex storefronts forked from metaplex, allowing for some customizations, improvements, and a wide range of sub-domain specific customizations; nowadays the source code for both metaplex and holaplex/metaplex is a little bit different, although a lot of the original parts are still being pretty similar.

Most recently, metaplex implemented this functionality, allowing for people to list auctions giving specific tokens, but since we forked such a long time ago we may need to do a lot of changes if we just try to merge with upstream.

### Development Work

The Solana programs that are used by Holaplex are compatible with using custom SPL tokens, which scopes the work that needs to be done as Frontend only; A realistic set of work to implement this includes: 

1. Given the reference commits and pull-requests for the work done on implementing custom SPL-Auctions, it would be a good idea and reference implementation.
2. Once implemented, we should be able to run the following, not-limited-to tests:
  - Instant Sale Auctions.
  - Auctions With Multiple Winners (Limited Edition Amount).
  - Regular Auctions.
  - Auctions with buyable participation NFTs.

**Comment by dandlezzz** [comment]
This looks close! https://github.com/holaplex/metaplex/pull/346
